### PR TITLE
Add doc strings (#7)

### PR DESCRIPTION
**Pull request by swadeley**
Hello

Please review, possibly even accept.

Thank you

### CODE REVIEW

**Comment by swadeley** [comment]
I fixed many flake8 fails, but left these two:
content-host-helpers]$ flake8 container-context-manager.py
container-context-manager.py:1:1: F401 'random' imported but unused
container-context-manager.py:65:9: F821 undefined name 'selfParameters'

File: container-context-manager.py [hunk 1]
```diff
@@ -37,44 +62,76 @@ def _create(self):
             command="tail -f /var/log/rhsm/rhsm.log",
         )
         self._client.start(container=self._inst)
-        self.name = self.execute("hostname")
+        selfParameters.name = self.execute("hostname")
```
Comment (JacobCallahan) [inline container-context-manager.py]: what is this?

File: container-context-manager.py [hunk 2]
```diff
@@ -148,6 +211,16 @@ def install_katello_agent(self):
...
 class ContainerContext:
+    """A context manager for managing containers.
+
+    Attributes:
+        runtime (init): The Linux container runtime to use. Defaults to docker.
+        image (init): The name of the container image to build from.
+        tag (init): The tag of the operating system to use.
+        count (init): The number to containers to build. Defaults to one.
+        agent (boolian): If set to True, prepares container for katello-agent.
+
```
Comment (JacobCallahan) [inline container-context-manager.py]: might want to put the types of output you can get, depending on inputs.
```
returns:
    A container object, if count == 1 and there is one tag.
    A list of container objects, if count > 1 and there is one tag
    A dict with tags as keys and the above as values
```
Something like that.

File: container-context-manager.py (hunk 1 above)
Comment (swadeley) [reply container-context-manager.py]: Oh dear, that looks like a bad case of copy paste.

**Pull request merged**